src/ssl.c

```diff
@@ -3464,13 +3465,35 @@ static int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
              /* Make sure previous is free'd */
             if (ssl->buffers.weOwnCert) {
                 FreeDer(&ssl->buffers.certificate);
+                #ifdef OPENSSL_EXTRA
+                    FreeX509(ssl->ourCert);
+                    if (ssl->ourCert) {
+                        XFREE(ssl->ourCert, ssl->heap, DYNAMIC_TYPE_X509);
+                    }
+                #endif
             }
             XMEMCPY(&ssl->buffers.certificate, &der, sizeof(der));
+            #ifdef OPENSSL_EXTRA
+                ssl->ourCert = wolfSSL_X509_d2i(NULL,
```

Comment:
Does this processing have to be done now?  That is, could we wait to allocate and process ourCert until it's actually needed?  What about the 99% of people who won't use this?  Isn't that a waste of resources?